src/test/java/com/example/spring/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(
+        scripts = {
+                "classpath:database/shopping.carts/clear-cart.sql",
+                "classpath:database/cart.items/clear-cart-items.sql",
+                "classpath:database/books/clear-books.sql",
+                "classpath:database/shopping.carts/add-cart.sql",
+                "classpath:database/roles/add-role.sql",
+                "classpath:database/users-roles/add-user-role.sql",
+                "classpath:database/books/add-book.sql",
+                "classpath:database/cart.items/add-cart-items.sql",
+        },
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD
+)
+@Sql(
+        scripts = {
+                "classpath:database/users-roles/clear-users-roles.sql",
+                "classpath:database/roles/clear-role.sql",
+                "classpath:database/cart.items/clear-cart-items.sql",
+                "classpath:database/shopping.carts/clear-cart.sql",
+                "classpath:database/books/clear-books.sql"
+        },
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+)
+public class ShoppingCartControllerTest {
+    protected static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @DisplayName("Get the shopping cart for a user")
+    void getCart_ValidUser_Success() throws Exception {
+        // Given
+        String jwtToken = generateJwtTokenForTestUser();
+
+        // When
+        MvcResult result = mockMvc.perform(get("/cart")
+                        .header("Authorization", "Bearer " + jwtToken)
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        // Then
+        ShoppingCartDto cartDto = objectMapper.readValue(
+                result.getResponse().getContentAsString(), ShoppingCartDto.class);
+        assertNotNull(cartDto);
+    }
+
+    private String generateJwtTokenForTestUser() {
+        return jwtUtil.generateToken("testUser@test.com");
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
```

Comment:
Like here ) Why do you use different types of authorization in other test cases?